public/components/trace_analytics/components/traces/trace_table_helpers.tsx

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiButtonIcon,
+  EuiCopy,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLink,
+  EuiTableFieldDataColumnType,
+  EuiText,
+  EuiToolTip,
+} from '@elastic/eui';
+import { round, truncate } from 'lodash';
+import moment from 'moment';
+import React from 'react';
+import { TRACE_ANALYTICS_DATE_FORMAT } from '../../../../../common/constants/trace_analytics';
+import { TraceAnalyticsMode, TraceQueryMode } from '../../../../../common/types/trace_analytics';
+import { nanoToMilliSec } from '../common/helper_functions';
+
+export const fetchDynamicColumns = (columnItems: string[]) => {
+  return columnItems
+    .filter((col) => col.includes('attributes') || col.includes('instrumentation'))
+    .map((col) => ({
+      className: 'attributes-column',
+      field: col,
+      name: (
+        <EuiText className="euiTableCellContent">
+          <EuiToolTip content={col}>
+            <p className="euiTableCellContent__text attributes-column-header">{col}</p>
+          </EuiToolTip>
+        </EuiText>
+      ),
+      align: 'right',
+      sortable: true,
+      truncateText: true,
+      render: (item) =>
+        item ? (
+          <EuiText>
+            <EuiToolTip content={item}>
+              <EuiText size="s">
+                {item.length < 36 ? item : <div title={item}>{truncate(item, { length: 36 })}</div>}
```

Comment:
use css to truncate if possible, like https://github.com/opensearch-project/dashboards-observability/blob/fc78c82795f5d8c138f424ccf6c7f96eca09e2c1/public/components/trace_analytics/index.scss#L75-L82

Reply:
makes sense, removed lodash trucate usage in this file.